source/source_integration_test.go

```diff
@@ -211,6 +216,95 @@ func TestSource_Snapshot_Continue(t *testing.T) {
 	is.NoErr(err)
 }
 
+func TestSource_CDC_Success(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	ctx := context.Background()
+
+	tableName := "conduit_source_cdc_integration_test_success"
+
+	cfg := prepareConfig(t, tableName)
+	cfg[ConfigKeyColumns] = "int_column,text_column"
+
+	err := prepareData(
+		ctx, cfg[config.KeyAddress], cfg[config.KeyKeyspace], cfg[config.KeyTabletType], tableName, true,
+	)
+	is.NoErr(err)
+
+	t.Cleanup(func() {
+		err = clearData(ctx, cfg[config.KeyAddress], cfg[config.KeyKeyspace], cfg[config.KeyTabletType], tableName)
+		is.NoErr(err)
+	})
+
+	s := new(Source)
+
+	err = s.Configure(ctx, cfg)
+	is.NoErr(err)
+
+	// wait a bit for Vitess to get the "create table" query results from each shard
+	time.Sleep(time.Second * 3)
+
+	// Start first time with nil position.
+	err = s.Open(ctx, nil)
+	is.NoErr(err)
+
+	// switch to CDC iterator
+	_, err = s.Read(ctx)
+	is.Equal(err, sdk.ErrBackoffRetry)
+
+	err = insertRow(ctx, cfg[config.KeyAddress], cfg[config.KeyKeyspace], cfg[config.KeyTabletType], tableName, 1, "Bob")
+	is.NoErr(err)
+
+	time.Sleep(time.Second * 5)
+
+	readCtx, cancel := context.WithTimeout(ctx, time.Second*15)
+	defer cancel()
+
+	record, err := readWithRetry(readCtx, t, s, time.Second*5)
```

Comment:
Is it possible run read function without retry?

Reply:
Yeah, but we can get `ErrBackoffRetry`, that's why we need the retries

Reply:
I mean data from Vitess may not come in real-time, and delays may occur